presto-main/src/main/java/com/facebook/presto/execution/SqlQueryManager.java

```diff
@@ -365,6 +372,26 @@ public void enforceMemoryLimits()
     }
 
     /**
+     * Prune extraneous info from old queries
+     */
+    public void pruneExpiredQueries()
```

Comment:
This method (and removeExpiredQueries) should probably be private. It's not safe to let anyone other than the single-threaded executor task call it due to the fact that they cause or are sensitive to the queue to shrinking/growing.